### PR TITLE
CNV-42626: Transition kubevirt vms to use topology spread constraints

### PR DESCRIPTION
**Pull request by davidvossel**
The descheduler has a policy that knows how to redistribute pods based on their TopologySpreadConstraints. by transitioning VMs from pod anti-affinity to use topo spread constraints, we can integrate better with descheduler logic that will accurately re-distribute VMs after a mgmt/infra cluster is updated.

### CODE REVIEW

**Comment by davidvossel** [comment]
/hold

I need to test the update path to ensure this doesn't break updates in an unexpected way

**Review by stevekuznetsov — APPROVED** [review]
Logic here seems correct, but I can't speak to the implications on upgrade.

**Comment by davidvossel** [comment]
@stevekuznetsov can i get another review when you get a chance. I had to rebase

**Pull request merged**
